src/NotepadNext/dialogs/MainWindow.cpp

```diff
@@ -1854,7 +1854,15 @@ bool MainWindow::checkFileForModification(ScintillaNext *editor)
     }
     else if (state == ScintillaNext::Modified) {
         qInfo("ScintillaNext::Modified");
-        editor->reload();
+        const QString filePath = editor->getFilePath();
+        auto reply = QMessageBox::question(this, tr("Reload"), tr("\"%1\"<br><br>This file has been modified by another program.<br>Do you want to reload it?").arg(filePath));
```

Comment:
To better align to the other message boxes shown by the MainWindow I would suggest something like:

```cpp
QMessageBox::question(this, tr("Reload File"), tr("<b>%1</b> has been modified by another program. Do you want to reload it?").arg(filePath))
```